server/src/storage/datastore.ts

```diff
@@ -18,6 +18,13 @@ import {Datastore, Transaction} from '@google-cloud/datastore';
 import {Entity} from '@google-cloud/datastore/build/src/entity';
 
 import {Filter, OrderRule, StringKeyObject} from '../interfaces';
+import {
+  NotFoundError,
+  InternalServerError,
+  ConflictError,
+  HttpError,
```

Comment:
ideally, at the storage layer there is no concept of http error. storage error should be thrown and got converted into http error in the upper layer. wdyt?

Reply:
@ifhuang I have the same thoughts too! But when I was implementing it, I'm not sure if I want to add that additional layer of complexity haha :p I'll take another look into this!